File: ControlApi/SharedLibrary/DataTypes/Events/EventDataType.cs
using DataAccessLibrary.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SharedLibrary.DataTypes.Events
{
    public record EventDataType
    {
        public Guid Id { get; init; }

        public string Name { get; init; }

        public DateTime StartDate { get; init; }

        public DateTime EndDate { get; init; }

        public TimeSpan StartTime { get; init; }

        public TimeSpan EndTime { get; init; }

        public bool Monday { get; init; }

        public bool Tuesday { get; init; }

        public bool Wednesday { get; init; }

        public bool Thursday { get; init; }

        public bool Friday { get; init; }

        public bool Saturday { get; init; }

        public bool Sunday { get; init; }

        public RecurrencyType RecurrencyType { get; init; }
    }
}

File: FrontEnd/src/app/models/EventDataType.ts
export class Time {
    hours: number;
    minutes: number;
    seconds: number;

    constructor(hours: number, minutes: number, seconds: number) {
        this.hours = hours;
        this.minutes = minutes;
        this.seconds = seconds;
    }
}

export enum RecurrencyType {
    UNIQUE,
    YEARLY,
    MONTHLY,
    WEEKLY
}

export class EventDataType {
    id: string;
    name: string;
    buildingId: string;
    startDate: Date;
    endDate: Date;
    startTime: any;
    endTime: any;
    monday: boolean;
    tuesday: boolean;
    wednesday: boolean;
    thursday: boolean;
    friday: boolean;
    saturday: boolean;
    sunday: boolean;
    recurrencyType: RecurrencyType;
}